### DISCUSSION TITLE
How to handle Safari back and forward maintaining socket connection?

### DISCUSSION THREAD

**Discussion by minchanhan** [socketio/socket.io#4994] [Q&A]
I have a simple web game where the user automatically connects to a socket upon entering page, and they should disconnect when they close the tab/window, or move back/forward in browser. Each of the listed methods trigger the `beforeunload` event in my React app, **except for Safari's back/forward buttons**.

I'm not sure how to handle socket disconnects then, because if a user "leaves", it should trigger an event for all users in the same room as them. How can I make a user disconnect from the socket and "leave" the room even if they exit through Safari back/forward buttons?

**Comment by darrachequesne** [comment]
Could you please check with the `closeOnBeforeunload` set to `true`?

```js
const socket = io({
  closeOnBeforeunload: true
});
```

Reference: https://socket.io/docs/v4/client-options/#closeonbeforeunload

**Reply by minchanhan** [reply]
@darrachequesne This didn't change anything. I'm also using the ("are you sure that you want to leave this page?") dialog on beforeunload.
